_posts/en/newsletters/2021-06-30-newsletter.md

```diff
@@ -0,0 +1,172 @@
+
+- [Are the witnesses segregated with schnorr signatures?]({{bse}}106930)
+  Dalit Sairio asks why, since [schnorr signatures][topic schnorr signatures] do
+  not suffer from the same malleability that ECDSA signatures do, schnorr
+  signatures will still be segregated? Darosior points out that malleability is
+  only one of the many benefits of segwit and that signature malleability is
+  only part of broader script malleability.
+
+- [Possible amount of signatures with MuSig?]({{bse}}106929)
+  Nickler explains that for both [MuSig][topic musig] and MuSig2 the number of signers is
+  practically infinite, noting that in [his benchmark][nickler musig], 1 million
+  signers takes about 130 seconds on his laptop.
+
+- [Support for P2WSH-wrapped P2TR addresses?]({{bse}}106706)
+  In addition to [BIP341][bip341 p2sh footnote]'s collision security concern,
+  jnewbery also points out the privacy issue with having an additional output
+  type, and the questionable need for P2SH-wrapped taproot outputs given wide
+  ecosystem adoption of bech32 already.
+
+## Preparing for taproot #2: is taproot even worth it for single-sig?
+
+*A weekly series about how developers and service providers can prepare
+for the upcoming activation of taproot at block height {{site.trb}}.*
+
+{% include specials/taproot/en/01-single-sig.md %}
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+FIXME:harding (nothing yet as of Friday; will update Tuesday)
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
+repo], [Hardware Wallet Interface (HWI)][hwi repo],
+[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
+[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
+BOLTs][bolts repo].*
+
+- [Bitcoin Core #22154][] adds code that will allow the user to generate
+  [bech32m][topic bech32] addresses for P2TR scripts after taproot
+  activates in block {{site.trb}}, e.g. by calling `getnewaddress ""
+  bech32m`.  The feature only applies to wallets with taproot
+  descriptors (see [Newsletter #152][news152 p2tr descriptors]).  If a
+  transaction includes any bech32m addresses after taproot activation,
+  the descriptor wallet will also use a P2TR change output.
+
+- [Bitcoin Core #22166][] adds support for inferring tr() descriptors
+  from outputs, completing basic taproot descriptor support. While
+  descriptors are a compact encoding for a set of addresses, this allows
+  the wallet to look up the descriptor from an address and the
+  descriptor expansion. Descriptor inference is used to provide more
+  accurate information in responses to RPC calls such as `listunspent`.
```

Comment:
Ran it past Pieter, who corrected my description for accuracy, but perhaps it is needlessly fine-grained and it would be kinder to our readers to just give the need-to-know version:

```suggestion
- [Bitcoin Core #22166][] adds support for inferring tr() descriptors
  from outputs, completing basic taproot descriptor support.
```

Reply:
I'm splitting the difference, including all the text from your short version and some of the text from your long version.